packages/components/src/tabs/tablist.tsx

```diff
@@ -17,19 +17,36 @@ import type { TabListProps } from './types';
 import { useTabsContext } from './context';
 import { TabListWrapper } from './styles';
 import type { WordPressComponentProps } from '../context';
+import type { CSSProperties } from 'react';
 
 export const TabList = forwardRef<
 	HTMLDivElement,
 	WordPressComponentProps< TabListProps, 'div', false >
 >( function TabList( { children, ...otherProps }, ref ) {
 	const context = useTabsContext();
+
+	const [ indicatorPosition, setIndicatorPosition ] = useState( {
+		left: 0,
+		width: 0,
+	} );
+	const selectedId = context?.store.useState( 'selectedId' );
+	const selectedTabEl = context?.store.item( selectedId )?.element;
+
+	useEffect( () => {
+		if ( selectedTabEl )
+			setIndicatorPosition( {
+				left: selectedTabEl.offsetLeft,
+				width: selectedTabEl.getBoundingClientRect().width,
+			} );
+	}, [ selectedTabEl ] );
```

Comment:
This is not going to be enough because the selected tab width can change dynamically.

You can see this kind of behavior in the block editor, for example:

https://github.com/WordPress/gutenberg/assets/555336/de3e3eec-12ac-45fb-8a22-7b0aada2c36f

We have a [`ResizeObserver` utility hook](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-compose/#useresizeobserver) that could be used.

Reply:
Implemented using ResizeObserver (not the linked hook because their API does not work here).